lockfree/src/list.rs

```diff
@@ -79,10 +79,9 @@ where
                         prev_next = next;
                     }
                 }
-                (eq, 0) => {
-                    next = curr_node.next.load(Ordering::Acquire, guard);
-                    if next.tag() == 0 {
-                        break eq == Equal;
+                (cmp, 0) => {
+                    if curr_node.next.load(Ordering::Relaxed, guard).tag() == 0 {
```

Comment:
Why you load `curr_node` once again?

Reply:
Why it can be relaxed?

Reply:
* I decided not to touch it because Harris' original algorithm does it implicitly but I think it can be removed.
* Reloading is sorely for checking if another thread logically removed `curr_node`. So there is no synchronization related to `curr_node.next` at this point.